File: SwiftAssignments/SwiftAssignments/Questions/K01.swift
import Foundation

class K01 : Question {
    func question1() {
        print("Hello World! ようこそSwift言語の世界へ！")
    }

    func question2() {
        print("Hello World!")
        print("ようこそ")
        print("Swift言語の世界へ！")
    }

    func question3() {
        print("整数：\(12345)")
        print("実数：\(123.456789)")
        print("文字：\("A")")
        print("文字列：\("ABCdef")")
    }

    func question4() {
        print("              ##")
        print("             #  #")
        print("             #  #")
        print("            #    #")
        print("           #      #")
        print("         ##        ##")
        print("       ##            ##")
        print("    ###                ###")
        print(" ###       ##    ##       ###")
        print("##        #  #  #  #        ##")
        print("##         ##    ##         ##")
        print(" ##     #            #     ##")
        print("  ###     ##########     ###")
        print("     ###              ###")
        print("        ##############")
    }

    func execute(number: Int) {
        switch number {
        case 1:
            question1()
        case 2:
            question2()
        case 3:
            question3()
        case 4:
            question4()
        default:
            break
        }
    }


}

File: GoRewrite/questions/k01.go
package questions

import "fmt"

type K01 struct {}

func (k K01) question1() {
	fmt.Println("Hello World! ようこそGo言語の世界へ！")
}

func (k K01) question2() {
	fmt.Println("Hello World!")
	fmt.Println("ようこそ")
	fmt.Println("Go言語の世界へ！")
}

func (k K01) question3() {
	fmt.Printf("整数：%d\n", 12345)
	fmt.Printf("実数：%f\n", 123.456789)
	fmt.Printf("文字：%c\n", 'A')
	fmt.Printf("文字列：%s\n", "ABCdef")
}

func (k K01) question4() {
	fmt.Println("              ##")
	fmt.Println("             #  #")
	fmt.Println("             #  #")
	fmt.Println("            #    #")
	fmt.Println("           #      #")
	fmt.Println("         ##        ##")
	fmt.Println("       ##            ##")
	fmt.Println("    ###                ###")
	fmt.Println(" ###       ##    ##       ###")
	fmt.Println("##        #  #  #  #        ##")
	fmt.Println("##         ##    ##         ##")
	fmt.Println(" ##     #            #     ##")
	fmt.Println("  ###     ##########     ###")
	fmt.Println("     ###              ###")
	fmt.Println("        ##############")
}

func (k K01) Execute(number int) {
	switch number {
	case 1:
		k.question1()
	case 2:
		k.question2()
	case 3:
		k.question3()
	case 4:
		k.question4()
	}
}

